### PR TITLE
redumped and reconverted CDs for Keyboardmania and Keyboardmania 2nd Mix + new clone [Jennifer Taylor]

### PR DESCRIPTION
**Pull request by DavidHaywood**
- redumped and reconverted CDs for Keyboardmania and Keyboardmania 2nd Mix [Jennifer Taylor]

new NOT WORKING clones
-----
Keyboardheaven (Korea) [Jennifer Taylor]

### CODE REVIEW

File: src/mame/drivers/firebeat.cpp [hunk 1]
```diff
@@ -2312,9 +2326,10 @@ GAME( 2000, ppd,    0,   firebeat_ppp, ppp, firebeat_ppp_state, init_ppd, ROT0,
 GAME( 2000, ppp11,  0,   firebeat_ppp, ppp, firebeat_ppp_state, init_ppp, ROT0, "Konami", "ParaParaParadise v1.1", MACHINE_NOT_WORKING )
 GAME( 2000, ppp1mp, ppp, firebeat_ppp, ppp, firebeat_ppp_state, init_ppp, ROT0, "Konami", "ParaParaParadise 1st Mix Plus", MACHINE_NOT_WORKING )
 
-GAMEL( 2000, kbm,    0, firebeat_kbm, kbm, firebeat_kbm_state, init_kbm, ROT270, "Konami", "Keyboardmania", MACHINE_NOT_WORKING, layout_firebeat )
-GAMEL( 2000, kbm2nd, 0, firebeat_kbm, kbm, firebeat_kbm_state, init_kbm, ROT270, "Konami", "Keyboardmania 2nd Mix", MACHINE_NOT_WORKING, layout_firebeat )
-GAMEL( 2001, kbm3rd, 0, firebeat_kbm, kbm, firebeat_kbm_state, init_kbm, ROT270, "Konami", "Keyboardmania 3rd Mix", MACHINE_NOT_WORKING, layout_firebeat )
+GAMEL( 2000, kbm,    0,   firebeat_kbm, kbm, firebeat_kbm_state, init_kbm, ROT270, "Konami", "Keyboardmania", MACHINE_NOT_WORKING, layout_firebeat )
+GAMEL( 2000, kbh,    kbm, firebeat_kbm, kbm, firebeat_kbm_state, init_kbm, ROT270, "Konami", "Keyboardheaven (Korea)", MACHINE_NOT_WORKING, layout_firebeat ) // gives E940 Hardware Error
```
Comment (987123879113) [inline src/mame/drivers/firebeat.cpp:2330]: I don't have the data so I can't test for myself, but you most likely need to add a new `init_kbh` and set `m_cabinet_info` to 3 (bit 0 is the region bit).

See: `init_ppp` (Japanese) and `init_ppd` (Korean).

The Japanese dongle `gq974-ja` also most likely won't work with the Korean version and would need to be dumped from a real dongle or generated using the internal sector keys inside the game's executables. And depending on the game the dongle serial needs to be set a certain way for "overseas" type dongles.

Comment (987123879113) [reply src/mame/drivers/firebeat.cpp:2330]: Assuming that the sector keys didn't change between Keyboardmania and Keyboard Heaven, copy `gq974-ja` to `gq974-ka` and modify the the last 0x30 bytes to say `MoDe:OVeRSea/PRoGRaMeD BY USR01979(HaJiMe SaiTo)`. 

The resulting SHA-1 should be `C3D07AAADF3D09FA412172EAF2C8475C342D94E7` if done correctly (using the latest `gq974-ja` I submitted to the mamedev email the other week).

You *may* need to change the serial at offset 0x13 from `1` to `2`.

This is all just guessing though.

Comment (DavidHaywood) [reply src/mame/drivers/firebeat.cpp:2330]: with the region change it boots, and seems happy with the security

Comment (987123879113) [reply src/mame/drivers/firebeat.cpp:2330]: Some games can be picky about the dongle data but others not so much, so if it's booting with just changing the region bit then great. :D

File: src/mame/mame.lst [hunk 2]
```diff
@@ -13782,6 +13782,7 @@ bm37th                          // GCB07 (c) 2002 (Japan JCA)
 bm3core                         // GCA05 (c) 2000 (Japan)
 bm3final                        // GCC01 (c) 2003 (Japan JCA)
 kbm                             // GQ974 (c) 2000 (Japan)
+kbh                             // GU974 (c) 2000 (Korea)
```
Comment (cuavas) [inline src/mame/mame.lst:13785]: Please keep this in alphabetical order.

Comment (DavidHaywood) [reply src/mame/mame.lst:13785]: it's a clone, should the clone really be listed first?

Comment (cuavas) [reply src/mame/mame.lst:13785]: The `.lst` and `.flt` files are just sorted for the most part to make them more manageable.

**Pull request merged**